File: Python/5.에라토스테네스의 체/1929.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/1929
'''

n, m = map(int, input().split())
l = [True] * 1000001

l[1] = False
for i in range(2, 1000001) :
    if l[i] == True :
        for j in range(i*2, 1000001, i) : 
            l[j] = False

for i in range(n, m+1) :
    if l[i] == True :
        print(i)

File: C++/7.1 에라토스테네스의 체/1929.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/1929
*/

#include <iostream>
#include <algorithm>
#include <cmath>

using namespace std;

int m, n;
bool check[1000001];

int main(){
    
    
    cin >> m >> n;
    
    fill(check, check+1000001, true);
    check[1] = false;
    
    for(int i=2; i<=sqrt(n); i++){
        if(check[i]){
            for(int j=i+i; j<=n; j+=i){
                check[j] = false;
            }
        }
    }
    
    for(int i=m; i<=n; i++){
        if(check[i]){
            cout << i << "\n";
        }
    }
    
    return 0;
}